core/cache/cachetest/state.go

```diff
@@ -32,13 +35,22 @@ func ModelChange(c *gc.C, model *state.Model) cache.ModelChange {
 	status, err := model.Status()
 	c.Assert(err, jc.ErrorIsNil)
 
+	users, err := model.Users()
+	permissions := make(map[string]permission.Access)
+	for _, user := range users {
+		// Cache permission map is always lower case.
+		permissions[strings.ToLower(user.UserName)] = user.Access
```

Comment:
This works for external users also? ie fred@external.
I can't recall if UserName includes the domain suffix

Reply:
It does. UserName is the fred@external bit.